allauth/socialaccount/providers/apple/views.py

```diff
@@ -112,19 +130,30 @@ class AppleOAuth2CallbackView(AppleOAuth2ClientMixin, OAuth2CallbackView):
         * Apple requests callback by POST
     """
 
-    def dispatch(self, request, *args, **kwargs):
-        if request.method == "POST":
-            url = request.build_absolute_uri(request.get_full_path())
-            params = {
-                "code": request.POST.get("code"),
-                "state": request.POST.get("state"),
-            }
-            return HttpResponseRedirect("%s?%s" % (url, urlencode(params)))
-        if request.method == "GET":
-            return super().dispatch(request, *args, **kwargs)
+    def get_user_scope_data(self):
+        user_scope_data = get_request_param(self.request, "user")
+        try:
+            return json.loads(user_scope_data)
+        except json.JSONDecodeError:
+            # We do not care much about user scope data as it maybe blank
+            # so return blank dictionary instead
+            return {}
+
+    def get_identity_data(self):
+        id_token = get_request_param(self.request, "id_token")
+        identity_token_data = self.adapter.get_verified_identity_data(id_token=id_token)
+
+        identity_token_data["user_scope_data"] = self.get_user_scope_data()
+
+        return identity_token_data
+
+    def get_token_data(self, app):
+        client = self.get_client(self.request, app)
+        identity_data = self.get_identity_data()
+        access_token_data = client.get_access_token(self.code)
```

Comment:
The fun thing here, is that we don't actually _need_ to get the access_token anymore. All the info we need is provided in the `form_post` response.

I was thinking about suggesting that we skip it, however if/when apple add some service which can be accessed with the `access_token` this code would need overhauling again, so probably best to leave as is

Reply:
@benhowes Apple says that we need to verify the user every 24 hours or so! I do not think it really make sense!